assembly/pallets/balances/assembly/balances-module.ts

```diff
@@ -0,0 +1,99 @@
+import { u128 } from "as-bignum";
+import { ByteArray, BytesReader, UInt64 } from "as-scale-codec";
+import { AccountData, ISignedTransaction, Log, ResponseCodes, Storage, TransactionValidity } from 'subsembly-core';
+import { System } from "../../../frame";
+import { AccountIdType, Balance } from "../../../runtime/runtime";
+
+/**
+ * @description The Balances Module.
+ * Used for account balance manipulation such as:
+ *  - Getting and setting free/reserved balances
+ */
+export class Balances {
+
+    /**
+     * @description Returns AccountData for a given AccountId
+     * If the account does not exist, Default AccountData is returned.
+     */
+    static getAccountData(accountId: AccountIdType): AccountData {
+        const accDataBytes = Storage.get(accountId.getAddress());
+        if (accDataBytes.isSome()) {
+            return BytesReader.decodeInto<AccountData>((<ByteArray>accDataBytes.unwrap()).values);
+        } else {
+            return AccountData.getDefault();
+        }
+    }
+
+    /**
+     * @description Sets the balances of a given AccountId
+     * Alters the Free balance and Reserved balances in Storage.
+     */
+    static setBalance(accountId: AccountIdType, freeBalance: Balance, reservedBalance: Balance): AccountData {
+        const currentAccountData = this.getAccountData(accountId);
+
+        // TODO Any meaningful checks
+
+        currentAccountData.setFree(freeBalance);
+        currentAccountData.setReserved(reservedBalance);
+        Storage.set(accountId.getAddress(), currentAccountData.toU8a());
+
+        return currentAccountData;
+    }
+
+    /**
+     * @description Transfer the given amount from sender to receiver
+     * Note: this is just draft implementation, without necessary checks
+     * @param sender sender account
+     * @param receiver receiver account
+     * @param amount amount of the transfer
+     */
+    static transfer(sender: AccountIdType, receiver: AccountIdType, amount: Balance): void {
+        const senderAccData = this.getAccountData(sender);
+        const receiverAccData = this.getAccountData(receiver);
+        const senderNewBalance: Balance = instantiate<Balance>(u128.sub(senderAccData.getFree().value, amount.value));
```

Comment:
I think that by doing `u128.sub` we are assuming that the balances module uses u128.
Maybe we should be able to do `Balance.sub(senderAccData.getFree()` in order not to assume the type